commands/new_function.go

```diff
@@ -140,23 +141,27 @@ func runNewFunction(cmd *cobra.Command, args []string) error {
 		}
 	}
 
-	if _, err := os.Stat(functionName); err == nil {
-		return fmt.Errorf("folder: %s already exists", functionName)
+	if len(handlerDir) == 0 {
+		handlerDir = functionName
```

Comment:
Hey there! I see only one problem here, every time we append new function with the current change we add `./` in front of the handler path so with the current change every time we use the `--append` flag in front of the `handler: ./<handler_folder>` in .yml file the `./` is appended so it becomes `././<handler_folder>` on 2nd append and `./././<handler_folder>` on 3rd append etc .. because if we look down we see `handler: ./{{ $function.Handler }}`.Got to find a way around that, but your code is very clean and good looking, great job!

Reply:
@martindekov is a change needed from @dmcclory ?

Reply:
Yes, with the normal faas-cli I don't have the same problem.

Reply:
Another way to go, is not to prefix the commandline `--handler` argument with `./` but default, if not given i.e.:
```golang
if len(handlerDir) == 0 {
	handlerDir = fmt.Sprintf("./%s", functionName)
}
```

And remove prefixing elsewhere i.e.

```golang
{{- range $name, $function := .Functions }}
  {{ $name }}:	  {{ $name }}:
    lang: {{ $function.Language }}
  handler: ./{{ $function.Handler }} <--- remove ./
    image: {{ $function.Image }}
{{- end }}	{{- end }}
```

```golang
if _, err := os.Stat("./" + dirName); os.IsNotExist(err) {      <--- remove ./
	t.Fatalf("%s/ directory was not created", dirName)
}
```
etc...

Reply:
Does it matter if much if the `handler` value in the yaml doesn't start with a `./`?

I was thinking of doing this:

```
  {{- range $name, $function := .Functions }}
    {{ $name }}:
      lang: {{ $function.Language }}
      handler: {{ $function.Handler }}
      image: {{ $function.Image }}
  {{- end }}
```

and just adding the prefix here:
```
    function := stack.Function{
      Name:     functionName,
      Handler:  "./" + handlerDir,
      Language: language,
      Image:    imageName,
    }
```